src/datasources/location-data-source.ts

```diff
@@ -0,0 +1,30 @@
+import { openmrsFetch, OpenmrsResource } from '@openmrs/esm-framework';
+import { DataSource } from '../api/types';
+
+export class LocationDataSource implements DataSource<OpenmrsResource> {
+  private initialUrl: string;
+  private url: string;
+
+  constructor(url: string) {
+    this.initialUrl = url;
+    this.url = url;
+  }
+
+  fetchData(searchTerm: string, config?: Record<string, any>): Promise<any[]> {
+    this.url = this.initialUrl;
```

Comment:
Are there cases where you envision the URL to change for this Datasource? Why parameterize it? Why not fuse it within assuming this Datasource will only be used for locations? Ideally, it should accept adhoc query params through configurations and the search term but I'm not sure I understand why we require a URL in the constructor. 

The parameter only makes sense for data sources whose URL might change like the base data source.